crates/cairo-lang-dojo/src/db.rs

```diff
@@ -1,29 +1,53 @@
 use std::sync::Arc;
 
-use cairo_lang_compiler::db::RootDatabase;
-use cairo_lang_lowering::db::LoweringGroup;
+use cairo_lang_compiler::db::RootDatabaseBuilder;
 use cairo_lang_plugins::get_default_plugins;
-use cairo_lang_semantic::corelib::get_core_ty_by_name;
-use cairo_lang_semantic::db::SemanticGroup;
-use itertools::Itertools;
+use cairo_lang_starknet::plugin::StarkNetPlugin;
 
 use crate::plugin::DojoPlugin;
 
-/// Returns a compiler database tuned to Dojo (e.g. Dojo plugin).
-pub fn get_dojo_database() -> RootDatabase {
-    let mut db_val = RootDatabase::default();
-    let db = &mut db_val;
-
-    // Override implicit precedence for compatibility with the Starknet OS.
-    db.set_implicit_precedence(Arc::new(
-        ["Pedersen", "RangeCheck", "Bitwise", "GasBuiltin", "System"]
-            .iter()
-            .map(|name| get_core_ty_by_name(db, name.into(), vec![]))
-            .collect_vec(),
-    ));
-
-    let mut plugins = get_default_plugins();
-    plugins.push(Arc::new(DojoPlugin {}));
-    db.set_semantic_plugins(plugins);
-    db_val
+pub trait DojoRootDatabaseBuilderEx {
+    /// Tunes a compiler database to Dojo (e.g. Dojo plugin).
+    fn with_dojo(&mut self) -> &mut Self;
```

Comment:
Right now we also append the impl uses in the plugins. It will change eventually.

Reply:
> looks like this isn't being used. is it useful to keep?

It is not currently used, but it could be used later, when the builder accepts several plugins in a row.